test/functional/rpc_fundrawtransaction.py

```diff
@@ -1200,6 +1201,59 @@ def test_add_inputs_default_value(self):
 
         self.nodes[2].unloadwallet("test_preset_inputs")
 
+    def test_preset_inputs_selection_crash(self):
+        self.log.info('Test wallet preset inputs selection')
+
+        # Here will confuse the wallet to make it select the
+        # preset inputs twice during the coin selection process (on v24.0).
+        #
+        # Making it think that the selection process result covers
+        # the entire tx target when it does not. It's actually creating
+        # a tx that sends more coins than what inputs are covering for.
+        #
+        # Which, combined with the subtract fee from outputs option,
+        # makes the wallet generate and broadcast a tx that pays
+        # up to the sum of all the preset inputs, minus one, amounts in fee!
+
+        # Create and fund the wallet with 2 UTXO of 5 BTC each
+        self.nodes[2].createwallet("test_preset_inputs_selection")
+        wallet = self.nodes[2].get_wallet_rpc("test_preset_inputs_selection")
+        outputs = {}
+        for _ in range(2):
+            outputs[wallet.getnewaddress(address_type="bech32")] = 5
+        self.nodes[0].sendmany("", outputs)
+        self.generate(self.nodes[0], 1)
+
+        # Select the preset inputs
+        coins = wallet.listunspent()
+        preset_inputs = [coins[0], coins[1]]
+
+        # Now let's create the tx
+        options = {
+            "inputs": [
+                {
+                    "txid": preset_inputs[0]["txid"],
+                    "vout": preset_inputs[0]["vout"]
+                },
+                {
+                    "txid": preset_inputs[1]["txid"],
+                    "vout": preset_inputs[1]["vout"]
+                },
+            ],
+            "add_inputs": True,
+            "subtract_fee_from_outputs": [0],
+            "add_to_wallet": False
+        }
+
+        # First case, use 'subtract_fee_from_outputs' to make the wallet create a tx that pays an insane fee on v24.0.
+        assert_raises_rpc_error(-4, "Insufficient funds", wallet.send, outputs=[{wallet.getnewaddress(address_type="bech32"): 14}], options=options)
+
+        # Second case, don't use 'subtract_fee_from_outputs' to make the wallet crash due the insane fee on v24.0
+        del options["subtract_fee_from_outputs"]
+        assert_raises_rpc_error(-4, "Insufficient funds", wallet.send, outputs=[{wallet.getnewaddress(address_type="bech32"): 14}], options=options)
```

Comment:
As it's written, the test comment doesn't make sense unless you have the background context. The comment describes a certain type of failure (insane fee, or wallet crash), but the assertion seems to check something entirely different. Perhaps it's worth writing a long comment which explains why you are checking for insufficient funds and what you would expect if the insufficient funds assert fails.